packages/server/src/queries/complex/pools/providers/sidecar.ts

```diff
@@ -27,51 +33,83 @@ const ExcludedExternalBoostPools: string[] =
 export function getPoolsFromSidecar({
   assetLists,
   poolIds,
+  notPoolIds,
+  types,
+  incentives,
   minLiquidityUsd,
   withMarketIncentives = true,
+  pagination,
+  sort,
+  search,
 }: {
   assetLists: AssetList[];
   chainList: Chain[];
   poolIds?: string[];
+  notPoolIds?: string[];
+  types?: PoolType[];
+  incentives?: string[];
   minLiquidityUsd?: number;
   withMarketIncentives?: boolean;
-}): Promise<Pool[]> {
-  if (poolIds && !poolIds.length) return Promise.resolve([]);
+  search?: SearchType;
+  pagination?: PaginationType;
+  sort?: SortType;
+}): Promise<PoolProviderResponse> {
+  if (poolIds && !poolIds.length) {
+    return Promise.resolve({ data: [], total: 0, nextCursor: undefined });
+  }
 
   return cachified({
     cache: poolsCache,
     key:
+      JSON.stringify(assetLists) +
```

Comment:
I think this can create some memory issues, because assetList is a very large array of object, if we need this maybe is better to just store asset unique identifier

Reply:
Good catch! No, it's not needed.